nautilus_core/model/src/data/bar.rs

```diff
@@ -310,8 +309,8 @@ impl Bar {
         volume: Quantity,
         ts_event: UnixNanos,
         ts_init: UnixNanos,
-    ) -> Self {
-        Self {
+    ) -> anyhow::Result<Self> {
+        Ok(Self {
```

Comment:
We probably don't gain anything by returning a result? because `Bar` creation is infallible thanks to the domain objects.